modules/kernel/src/main/scala/notebook/Messages.scala

```diff
@@ -18,6 +18,8 @@ sealed trait CalcResponse
 
 case class StreamResponse(data: String, name: String) extends CalcResponse
 
+case class DefinitionResponse(definedTermOrType: Option[Either[String, String]], tpe:String, cell:String, references: List[String]) extends CalcResponse
```

Comment:
yeah, like:
```scala
abstract class NameDefinition(name: String, tpe: String, references: List[String])
case class TypeDefinition(name: String, tpe: String, references: List[String]) extends NameDefinition
case class TermDefinition(name: String, tpe: String, references: List[String]) extends NameDefinition
```

another question is if we need to include `cell` field, if not, the `Repl.evaluate` can be simpler, and use the same ADT:

```scala
def evaluate( code: String,
              onPrintln: String => Unit = _ => (),
              onNameDefinition: NameDefinition => Unit  = _ => ()
              )
```